tensorflow/src/test/scala/magnolify/tensorflow/test/ExampleTypeSuite.scala

```diff
@@ -120,36 +120,46 @@ class ExampleTypeSuite extends MagnolifySuite {
   }
 
   test("WithAnnotations") {
-    implicit val et: ExampleType[WithAnnotations] = ExampleType[WithAnnotations]
+    implicit val ef: ExampleField[NestedWithAnnotations] = ExampleField.gen
+    val et = ExampleType[WithAnnotations]
+
+    val expectedFieldSchema = Schema
+      .newBuilder()
+      .addFeature(feature("b", FeatureType.INT, Some("other b doc")))
+      .addFeature(feature("i", FeatureType.INT, Some("other i doc")))
+      .setAnnotation(annotation("this doc will be ignored"))
+      .build()
 
     val expectedSchema = Schema
       .newBuilder()
-      .addFeature(noAnnotationFeature("noAnnotationB", FeatureType.INT))
-      .addFeature(feature("b", FeatureType.INT))
-      .addFeature(feature("i", FeatureType.INT))
-      .addFeature(feature("maybeI", FeatureType.INT))
-      .addFeature(feature("l", FeatureType.INT))
-      .addFeature(feature("f", FeatureType.FLOAT))
-      .addFeature(feature("d", FeatureType.FLOAT))
-      .addFeature(feature("bs", FeatureType.BYTES))
-      .addFeature(feature("s", FeatureType.BYTES))
-      .addFeature(feature("ii", FeatureType.INT))
-      .addFeature(feature("ff", FeatureType.FLOAT))
-      .addFeature(feature("bsbs", FeatureType.BYTES))
-      .addFeature(feature("nested.b", FeatureType.INT))
-      .addFeature(feature("nested.i", FeatureType.INT))
+      .addFeature(feature("noAnnotationB", FeatureType.INT, None))
+      .addFeature(feature("b", FeatureType.INT, Some("b doc")))
+      .addFeature(feature("i", FeatureType.INT, Some("i doc")))
+      .addFeature(feature("maybeI", FeatureType.INT, Some("maybeI doc")))
+      .addFeature(feature("l", FeatureType.INT, Some("l doc")))
+      .addFeature(feature("f", FeatureType.FLOAT, Some("f doc")))
+      .addFeature(feature("d", FeatureType.FLOAT, Some("d doc")))
+      .addFeature(feature("bs", FeatureType.BYTES, Some("bs doc")))
+      .addFeature(feature("s", FeatureType.BYTES, Some("s doc")))
+      .addFeature(feature("ii", FeatureType.INT, Some("ii doc")))
+      .addFeature(feature("ff", FeatureType.FLOAT, Some("ff doc")))
+      .addFeature(feature("bsbs", FeatureType.BYTES, Some("bsbs doc")))
+      .addFeature(feature("nested.b", FeatureType.INT, Some("other b doc")))
+      .addFeature(feature("nested.i", FeatureType.INT, Some("other i doc")))
       .setAnnotation(annotation("Example top level doc"))
       .build()
 
-    val s = et.schema
+    assertEquals(ef.schema(CaseMapper.identity), expectedFieldSchema)
     assertEquals(et.schema, expectedSchema)
   }
 
-  private def noAnnotationFeature(name: String, t: FeatureType): Feature =
```

Comment:
this might be personal preference so feel free to ignore it, but I think is more readable to have a dedicated method instead of passing an option argument